tests/shell_handler.test.ts

```diff
@@ -0,0 +1,23 @@
+import { assertEquals, Stub, stub } from "../dev_deps.ts";
+import { getShellConfigFullPath } from "../utils/shell_handler.ts";
+
+const mockDenoEnv = (denoEnvMock: string[] | null[]): Stub<
+  {
+    get(key: string): string | undefined;
+    set(key: string, value: string): void;
+    delete(key: string): void;
+    toObject(): { [index: string]: string };
+  }
+> =>
+  stub(Deno.env, "get", [
+    ...denoEnvMock,
+  ]);
+
+Deno.test("should return shell config full path", () => {
+  const denoEnvMock = ["zsh", "/Users/$USER"];
+  mockDenoEnv(denoEnvMock);
+  assertEquals(
+    getShellConfigFullPath(),
+    `${denoEnvMock[1]}/.${denoEnvMock[0]}rc`,
```

Comment:
With the mock, we succeeded to work around the `Deno.env.get` calls, but I would say that, if we use the mock to construct our assertion, we are not testing it correctly.

We need to mock the 2 calls of `Deno.env.get`, the first one for **SHELL** env and the last one for **HOME** env. So, our mock should be the value of `[ "/bin/zsh", "/Users/randomUser"]` and the second argument of `assertEquals` should be `/Users/randomUser/.zshrc`.

I played around a bit and came with this:

```ts
const get: Stub<typeof Deno.env> = stub(Deno.env, "get");

Deno.test("should return shell config full path if SHELL env is defined", () => {
  get.returns = ["/bin/bash", "/Users/randomUser"];
  assertEquals(
    getShellConfigFullPath(),
    `/Users/randomUser/.bashrc`,
  );
});

Deno.test("should return undefined if SHELL env is undefined", () => {
  assertEquals(
    getShellConfigFullPath(),
    undefined,
  );
});
```

What do you think about it?

Reply:
I would suggest initiating the Stub one with the values that will be our mocks. In other words, it'll be better if we create a callback that receives the mock and returns the Stub class.
Other than that, will be nice if we have a proper return in `getShellConfigFullPath`, because the second test that you suggest won't work, because today if we mock a null as the return of `Deno.env.get('SHELL')`, or something like that, we won't have a value in our return at all.
So, considering that, I guess the right way would be:

1. Create a function that receives as a parameter the mock value and returns the Stub.
2. Use that function to mock both calls that we make to `Deno.env.get`
3. Refactor `getShellConfigFullPath` to return null/false when we can't get something from `Deno.env.get`

What do you think?